sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -346,6 +346,42 @@ object RapidsPluginUtils extends Logging {
       loadExtensions(classOf[SparkPlugin], pluginClasses)
     }
   }
+
+  /**
+   * Extracts supported GPU architectures from the given properties file
+   */
+  private def getSupportedGpuArchitectures(propFileName: String): Set[Int] = {
+    val props = RapidsPluginUtils.loadProps(propFileName)
+    Option(props.getProperty("gpu_architectures"))
+      .getOrElse(throw new RuntimeException(s"GPU architectures not found in $propFileName"))
+      .split(";")
+      .map(_.toInt)
+      .toSet
+  }
+
+  /**
+   * Checks if the current GPU architecture is supported by the spark-rapids-jni
+   * and cuDF libraries.
+   *
+   * Compatibility Rule: Binaries generated for a certain compute capability is supported
+   * to run on any GPU with the same major revision and same or higher minor revision of
+   * compute capability.
+   *
+   * See: https://docs.nvidia.com/cuda/ampere-compatibility-guide/index.html
+   */
+  def checkGpuArchitecture(): Unit = {
+    val supportedGpuArchsSet = getSupportedGpuArchitectures(JNI_PROPS_FILENAME)
+      .intersect(getSupportedGpuArchitectures(CUDF_PROPS_FILENAME))
+    val majorGpuArch = Cuda.getComputeCapabilityMajor
+    val minorGpuArch = Cuda.getComputeCapabilityMinor
+    val isSupported = supportedGpuArchsSet.exists { ver =>
+      majorGpuArch == ver / 10 && minorGpuArch >= ver % 10
+    }
```

Comment:
Per the offline discussion, I think we want to not be explicit about major having to match since we're shipping PTX that can be used by the driver to JIT-compile for future architectures.  In that sense, we probably want to modify the check to something like this:
```suggestion
    val gpuArch = majorGpuArch * 10 + minorGpuArch
    val isSupported = supportedGpuArchsSet.exists(_ <= gpuArch)
```
And maybe we want to emit an additional warning on startup if the GPU is supported but we can't find a precompiled binary with the same major version.  That will require a potentially expensive JIT compile that might appear as a hang.